src/shared/mount-util.h

```diff
@@ -6,6 +6,25 @@
 
 #include "macro.h"
 
+/* 4MB for contents of regular files, 64k inodes for directories, symbolic links and device specials,
+   using large storage array systems as a baseline */
+#define TMPFS_LIMITS_DEV             ",size=4m,nr_inodes=64k"
+/* Very little, if any use expected */
+#define TMPFS_LIMITS_EMPTY_OR_ALMOST ",size=4m,nr_inodes=1k"
+#define TMPFS_LIMITS_SYS             TMPFS_LIMITS_EMPTY_OR_ALMOST
+#define TMPFS_LIMITS_SYS_FS_CGROUP   TMPFS_LIMITS_EMPTY_OR_ALMOST
+/* 10% of RAM (using 16GB of RAM as a baseline) translates to 400k inodes (assuming 4k each) and 25%
+   translates to 1M inodes */
+#define TMPFS_LIMITS_TMP             ",size=10%,nr_inodes=400k"
+#define TMPFS_LIMITS_DEV_SHM         TMPFS_LIMITS_TMP
+#define TMPFS_LIMITS_HOME            TMPFS_LIMITS_TMP
```

Comment:
We use this to block access to the real /home, and mount it read-only. In most ways this should be treated like ImmutablePath=, i.e. be bound to TMPFS_LIMITS_EMPTY_OR_ALMOST .

(I probably wouldn't define TMPFS_LIMITS_HOME at all, but just use TMPFS_LIMITS_EMPTY_OR_ALMOST in the protect_home_tmpfs_table[] table where it is used.)